### DISCUSSION TITLE
Will TS release the Worksheet Column Utilization liveboard as shown in the cs_tools highlights for v1.5.0?

### DISCUSSION THREAD

**Discussion by bmofkn** [thoughtspot/cs_tools#120] [Q&A] [answered]
This [page](https://thoughtspot.github.io/cs_tools/changelog/1.5.0/#__tabbed_1_2) in the [v1.5.0 change log](https://thoughtspot.github.io/cs_tools/changelog/1.5.0/) gives an example of three liveboards that would be extremely useful.

Are these example liveboards coming to the TS UI?  Or do we have to use `cs_tools tools searchable metadata` via a syncer to extract and load the data into TS (consuming additional row count) and then create our own liveboards?

![worksheet_column_utilization_overview](https://github.com/thoughtspot/cs_tools/assets/119008346/4323e326-264b-4386-b7b1-fd58eb1bf87f)

**Comment by bmofkn** [comment]
🤔 - perhaps this is the answer to my question... [searchable deploy](https://thoughtspot.github.io/cs_tools/tools/searchable/#__tabbed_1_4)?  If we are using the default cloud version of TS, is this as simple as supplying the default values for "falcon" as described in the docs?

**Reply by boonhapus** [reply, marked as answer]
Hey @bmofkn , you're correct it's the `searchable deploy` command. I'll take the implicit feedback that the docs should include a short guide on what the heck to even do with the commands. 😅

To get this Liveboard (and underlying Worksheets, Tables, etc) working.. you first hard to run `searchable metadata` **[with a database Syncer](https://thoughtspot.github.io/cs_tools/syncer/what-is/)** that ThoughtSpot supports connecting to. This will autocreate the tables in the database so that you can import the TML via the `searchable deploy` command.

For example.. 

```
cs_tools tools searchable metadata --syncer snowflake://account_name=...&username=...&secret=...&warehouse=WH_DATA_LOADS_XS&role=DATA_OPERATIONS&database=GO_TO_MARKET&authentication=basic

cs_tools tools searchable bi-server --from-date 2024-01-01 --to-date 2024-01-31 --syncer snowflake://account_name=...&username=...&secret=...&warehouse=WH_DATA_LOADS_XS&role=DATA_OPERATIONS&database=GO_TO_MARKET&authentication=basic&load_strategy=UPSERT
```

.. would load the both the metadata and activity data to my Snowflake database. The tables get created in `GO_TO_MARKET.PUBLIC`.

Then I would go create a new empty Connection in ThoughtSpot, connect it to the same schema, and finally run...

```
cs_tools tools searchable deploy --connection-guid 00000000-0000-0000-0000-000000000000 --database GO_TO_MARKET --schema PUBLIC
```

You can find your connection guid in the URL once you navigate to the Connections page in ThoughtSpot.

**Reply by bmofkn** [reply]
@boonhapus, we are on TS Cloud version `9.10.0.cl-188` and the source of our data is Redshift (... I don't see a syncer for Redshift... yet).

Sorry for my ignorance, but I'm guessing since we are using the cloud version, we cannot use the "Falcon" options with `searchable deploy` as the docs allude to?

`if Falcon, use falcon, otherwise find your guid in the Connection URL in the Data Workspace `

<img width="1061" alt="searchable deploy falcon" src="https://github.com/thoughtspot/cs_tools/assets/119008346/139bbed8-35c2-4c45-a52c-b39c425869f9">

While eager to see the Column Utilization Liveboard, we'll hold off until we know if this will affect our row count.  Please pass along to management that these cs_tools liveboards would be an excellent benefit for all customers within the Usage Statistics area without having to export/import via a syncer 🙂

**Reply by boonhapus** [reply]
"yet" indeed. 😄 Myself and another contributor are working on Redshift right now actually, with the intent to release it within the next week or so.

You're correct that Falcon isn't available in ThoughtSpot's cloud offering.

@bmofkn I confirmed that it would be included in your reported row count, as ThoughtSpot captures everything tied to your cluster id.. but still trying to nail down if we would factor this number into the licensing allowance, or discount/ignore it. I've posed the question to a few leaders but don't expect to have a complete answer* until next week.

**Comment by boonhapus** [comment]
To answer the original question of "Would it impact my licensing cost?" if you're on a row-based license.... let me ask internally @bmofkn . I am not entirely sure, but I should really know this. 👍

**Reply by boonhapus** [reply]
Hey @bmofkn , sorry this one took so long.

**_Officially..._** , if you are on a row-based or capacity licensing model, the rows you connect to your cluster will count towards your quota (even if it's based on ThoughtSpot data itself). There is no technical "shutoff valve" here, so if you were to connect 1T rows today, your cluster would still function appropriately.

Having said that, the licensing conversation is between you and your Account Executive at ThoughtSpot, and if you were to connect the data under a separate Connection named **ThoughtSpot Metadata**, and transfer the content to `tsadmin`, I'm sure you could negotiate that these rows should not count against your quota as it's used to support the health and use of your cluster much like the other system-level Worksheets. I unfortunately don't have any say here, so I would recommend reaching out to your AE to clarify this.
